### PR TITLE
Add endpoint to add a comment to an article

### PR DESCRIPTION
**Pull request by jasonbarrows**

### CODE REVIEW

**Comment by jasonbarrows** [comment]
Please excuse the mess on the timeline above.  I was agonising over whether to include custom validation to save trips to the database but have abandoned it for now.  This can be revisited after core tickets have been completed and will most likely be best solved with a package anyway.

File: __tests__/app.test.js [hunk 1]
```diff
@@ -185,3 +187,102 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+          comment_id: expect.any(Number),
+          body: expect.any(String),
+          article_id: expect.any(Number),
+          author: expect.any(String),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+        });
+      });
+  });
+
```
Comment (Jodiech) [inline __tests__/app.test.js:215]: I'd recommend adding another 201 for a test that ignores other additional properties that are passed 👍

File: endpoints.json [hunk 2]
```diff
@@ -64,5 +64,23 @@
...
+    "exampleResponse": {
+      "comment": {
+        "comment_id": 19,
+        "article_id": 1,
+        "body": "This article is most excellently written, if I should say so myself.",
+        "author": "butter_bridge",
+        "votes": 0,
+        "created_at": "2020-10-31T03:03:00.000Z"
+      }
+    }
```
Comment (Jodiech) [inline endpoints.json:84]: Nice one for updating the endpoints.json file 😄

File: __tests__/app.test.js [hunk 3]
```diff
@@ -185,3 +187,102 @@ describe('GET /api/articles/:article_id/comments', () => {
       });
   });
 });
+
+describe('POST /api/articles/:article_id/comments', () => {
+  it('200: creates a new comment for a valid article id and responds with the new comment object', () => {
```
Comment (Jodiech) [inline __tests__/app.test.js]: This should be a status 201 - for a created comment object 😉

**Review by Jodiech — APPROVED** [review]
Hey 👋 nice work on this ticket! Your tests are really thorough 🚀 happy for you to merge!

File: __tests__/app.test.js (hunk 3 above)
Comment (jasonbarrows) [reply __tests__/app.test.js]: Good catch!

File: __tests__/app.test.js (hunk 1 above)
Comment (jasonbarrows) [reply __tests__/app.test.js:215]: I've added a test for this and included `votes: 100, foo: 'bar'` in the request body with expectation for the following:

```javascript
expect(comment.votes).not.toBe(100);
expect(comment).not.toHaveProperty('foo');

**Pull request merged**